openassessment/assessment/test/test_staff.py

```diff
@@ -0,0 +1,348 @@
+            RUBRIC_DICT,
+        )
+        # Be sure to update the workflow!
+        workflow_api.update_from_assessments(tim_sub["uuid"], self.STEP_REQUIREMENTS, force_update_score=True)
+
+        # Verify both assessment and workflow report correct score
+        self.assertEqual(staff_assessment["points_earned"], 14)
+        workflow = workflow_api.get_workflow_for_submission(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+        self.assertEqual(workflow["score"]["points_earned"], 14)
+
+    def test_create_assessment_does_not_block(self):
+        # Create assessment
+        tim_sub, tim = self.create_student_and_submission("Tim", "Tim's answer")
+
+        # Staff assess it
+        staff_assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            ASSESSMENT_DICT_PASS['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+        # Keep the workflow updated
+        workflow_api.update_from_assessments(tim_sub["uuid"], self.STEP_REQUIREMENTS, force_update_score=True)
+
+        # Ensure points are what we expect
+        self.assertEqual(staff_assessment["points_earned"], 14)
+        workflow = workflow_api.get_workflow_for_submission(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+        self.assertEqual(workflow["score"]["points_earned"], 14)
+
+        # Now self assess, and ensure assessment is recorded
+        self_assessment = self_assess(
+            tim_sub["uuid"],
+            tim["student_id"],
+            ASSESSMENT_DICT['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+        self.assertEqual(self_assessment["points_earned"], 6)
+
+        # Ensure points are not updated
+        workflow = workflow_api.get_workflow_for_submission(tim_sub["uuid"], self.STEP_REQUIREMENTS)
+        self.assertEqual(workflow["score"]["points_earned"], 14)
+
+    def test_staff_assessment_required(self):
+        # Set up submission, with staff as a required step on the workflow
+        tim_sub, tim = self.create_student_and_submission("Tim", "Tim's answer", require_staff=True)
+
+        # Verify that we're still waiting on a staff assessment
+        self.assertFalse(staff_api.assessment_is_finished(tim_sub["uuid"], self.STEP_REQUIREMENTS_WITH_STAFF))
+
+        # Staff assess, without using force_update parameter
+        staff_assessment = staff_api.create_assessment(
+            tim_sub["uuid"],
+            "Dumbledore",
+            ASSESSMENT_DICT_PASS['options_selected'], dict(), "",
+            RUBRIC_DICT,
+        )
+        workflow_api.update_from_assessments(tim_sub["uuid"], self.STEP_REQUIREMENTS_WITH_STAFF)
+
+        # Verify assesment made, score updated, and no longer wating
+        self.assertEqual(staff_assessment["points_earned"], 14)
+        self.assertTrue(staff_api.assessment_is_finished(tim_sub["uuid"], self.STEP_REQUIREMENTS_WITH_STAFF))
+
+    def test_invalid_rubric_exceptions(self):
```

Comment:
I'm thinking it would be good to make this test use ``DDT`` so that it can validate a variety of different invalid options. Currently it is passing a whole set of invalid options at once, so it is hard to be sure that each different option is being validated. It could be that only one of them is failing, but the test still sees that as correct behavior.